frontend/src/ui-components/AcmSearchInput/AcmSearchInput.tsx

```diff
@@ -0,0 +1,273 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { useRef, useState, useEffect } from 'react'
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Flex,
+  FlexItem,
+  Form,
+  Panel,
+  PanelMain,
+  PanelMainBody,
+  Popper,
+  SearchInputProps,
+  SelectOption,
+} from '@patternfly/react-core'
+import { t } from 'i18next'
+import { AcmSelect } from '../AcmSelect'
+import { PlusCircleIcon, TimesCircleIcon } from '@patternfly/react-icons'
+import { AcmTextInput } from '../AcmTextInput'
+import { SearchInput } from './PFSearchInput'
+
+export enum SearchOperator {
+  Equals = '=',
+}
+
+export interface SearchableColumn {
+  columnId: string
+  availableOperators: SearchOperator[]
+  columnDisplayName?: string
+}
+
+export interface SearchConstraint {
+  columnId?: string
+  operator?: SearchOperator
+  value?: string
+}
+
+export interface AcmSearchInputProps extends SearchInputProps {
+  placeholder?: string
+  spellCheck?: boolean
+  resultsCount?: string | number
+  style?: React.CSSProperties
+  useAdvancedSearchPopper?: boolean
+  searchableColumns?: SearchableColumn[]
+  canAddConstraints?: boolean
+  disableAddConstraint?: boolean
+  pendingConstraints?: SearchConstraint[]
+  setActiveConstraints?: (constraints: SearchConstraint[]) => void
+  fuzzySearchOnChange?: (value: string) => void
+  fuzzySearchOnClear?: () => void
+  fuzzySearchValue?: string
+}
+
+export function AcmSearchInput(props: Readonly<AcmSearchInputProps>) {
+  const {
+    searchableColumns,
+    canAddConstraints,
+    disableAddConstraint,
+    pendingConstraints = [{ operator: SearchOperator.Equals, value: '', columnId: '' }],
+    fuzzySearchValue,
+    fuzzySearchOnChange,
+    fuzzySearchOnClear,
+    setActiveConstraints,
+    placeholder,
+    useAdvancedSearchPopper,
+    spellCheck,
+    resultsCount,
+    style,
+  } = props
+
+  const searchInputRef = useRef<HTMLInputElement | null>(null)
+  const advancedSearchPaneRef = useRef<HTMLDivElement | null>(null)
+  const [isAdvancedSearchOpen, setIsAdvancedSearchOpen] = useState(false)
+  const [searchConstraints, setSearchConstraints] = useState<SearchConstraint[]>(pendingConstraints || [])
+
+  const onClear = () => {
+    fuzzySearchOnClear && fuzzySearchOnClear()
```

Comment:
A bit more concise:
```suggestion
    fuzzySearchOnClear?.()
```
It might also be good practice to wrap these event handlers with `useCallback`... but it depends on whether anything in the components you are passing these to will get rerendered when they change.

Reply:
Had to take a moment to think about this, but yes, I suppose since they both have some control over the constraint array or the fuzzy search this makes sense...